### ISSUE TITLE
IMAPsync Two-way sync?

### ISSUE THREAD

**Issue by ChojinDSL** [mailcow/mailcow-dockerized#2326]
Would it be possible to add a option so that you can enable a two-way sync job? 
Essentially just a flag you check, and in that case the sync job is actually two sync jobs, one is server1->server2 and after that a sync job which does server2>server1. (I realize imapsync doesn't support a two sync directly)

I have a situation where my company has two offices. The one office definitely needs something like mailcow, since its primarily developers using linux and mac desktops. The other office are the "corporate suits" of the company and they rely on a local exchange server. 
Long story short, it's still going to take a while to migrate all the corporate users away from their exchange accounts. A two way Imap sync would go a long way to ensure that people still get all emails regardless of which server they land on.

I should also add that the exchange server is not under my control, and the corporate admin is less than cooperative. If it was under my control, this would be less of an issue.

**Comment by extremeshok** [comment]
A possible solution
Use Transport Maps to send the messages of the email addresses which are not on mailcow server to the exchange server
Then use the sync off the exchange server to the mailcow server for the email addresses which are on the mailcow server.

mailcow will have the dev1@, dev2@
exchange will have dev1@, dev2@, corp1@, corp2@

an email sent to corp1 from dev1 will be sent to the exchange server
an email sent to dev1 from dev2 will be sent via the mailcow server
and email sent to dev1 from corp2 will be sent via the exchange server, and fetched with the mailcoew server

**Comment by liquidat** [comment]
Maybe a different approach might be more useful here? How about using dovecot as an [IMAP proxy](https://wiki.dovecot.org/HowTo/ImapcProxy)? That way the mails are in the end all on the exchange server (or any other IMAP server in the end), but your users can use mailcow if wanted.
